msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfiguration.java

```diff
@@ -480,10 +483,11 @@ private void nullConfigurationCheck(String configKey, String configValue) {
     }
 
     private boolean isBrokerRedirectUri() {
-        final String BROKER_REDIRECT_URI_REGEX = "msauth://" + mAppContext.getPackageName() + "/.*";
-        final Pattern pairRegex = Pattern.compile(BROKER_REDIRECT_URI_REGEX);
-        final Matcher matcher = pairRegex.matcher(mRedirectUri);
-        return matcher.matches();
+        final Matcher matcher = BROKER_REDIRECT_URI_REGEX.matcher(mRedirectUri);
+        if (matcher.matches() && ObjectUtils.equals(mAppContext.getPackageName(), matcher.group(1))) {
```

Comment:
Optionally return the entire expression?

```java
return matcher.matches() && ObjectUtils.equals(mAppContext.getPackageName(), matcher.group(1));
```

Reply:
Yes, that's better.